lib/node-utils.js

```diff
@@ -244,6 +244,15 @@ function isComma(token) {
     return token.kind === SyntaxKind.CommaToken;
 }
 
+/**
+ * Returns true if the given TSToken is a comment
+ * @param  {TSToken}  token the TypeScript token
+ * @returns {boolean}       is commment
+ */
+function isComment(token) {
+    return token.kind === SyntaxKind.SingleLineCommentTrivia || token.kind === SyntaxKind.MultiLineCommentTrivia || (token.kind >= SyntaxKind.JSDocTypeExpression && token.kind <= SyntaxKind.JSDocTypeLiteral);
```

Comment:
Is this the correct check for comment tokens?

Reply:
TypeScript themselves use this within their private utilities:

```ts
function isComment(kind: SyntaxKind): boolean {
  return kind === SyntaxKind.SingleLineCommentTrivia || kind === SyntaxKind.MultiLineCommentTrivia;
}
```

I feel like we should split the JSDoc checks out into a separate function.

The TypeScript team has these (again not in their public API, but useful utils):

```ts
// JSDoc
    /** True if node is of some JSDoc syntax kind. */
    /* @internal */
    export function isJSDocNode(node: Node): boolean {
        return node.kind >= SyntaxKind.FirstJSDocNode && node.kind <= SyntaxKind.LastJSDocNode;
    }

    /** True if node is of a kind that may contain comment text. */
    export function isJSDocCommentContainingNode(node: Node): boolean {
        return node.kind === SyntaxKind.JSDocComment || isJSDocTag(node);
    }

    // TODO: determine what this does before making it public.
    /* @internal */
    export function isJSDocTag(node: Node): boolean {
        return node.kind >= SyntaxKind.FirstJSDocTagNode && node.kind <= SyntaxKind.LastJSDocTagNode;
    }
```